tests/integration/devfile/cmd_devfile_storage_test.go

```diff
@@ -26,373 +26,360 @@ var _ = Describe("odo devfile storage command tests", func() {
 		helper.CommonAfterEach(commonVar)
 	})
 
-	Context("When devfile storage create command is executed", func() {
+	When("creating a nodejs component", func() {
+		storageNames := []string{helper.RandString(5), helper.RandString(5)}
+		pathNames := []string{"/data", "/" + storageNames[1]}
+		sizes := []string{"5Gi", "1Gi"}
 
-		It("should create the storage and mount it on the container", func() {
+		BeforeEach(func() {
 			helper.Cmd("odo", "create", cmpName, "--context", commonVar.Context, "--project", commonVar.Project, "--devfile", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile.yaml")).ShouldPass()
-
 			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+		})
 
-			storageNames := []string{helper.RandString(5), helper.RandString(5)}
-			pathNames := []string{"/data", "/" + storageNames[1]}
-			sizes := []string{"5Gi", "1Gi"}
-
-			helper.Cmd("odo", "storage", "create", storageNames[0], "--path", pathNames[0], "--size", sizes[0], "--context", commonVar.Context).ShouldPass()
-			// check storage create without the path name
-			helper.Cmd("odo", "storage", "create", storageNames[1], "--size", sizes[1], "--context", commonVar.Context).ShouldPass()
-			helper.Cmd("odo", "push", "--context", commonVar.Context).ShouldPass()
+		It("should throw error if no storage is present", func() {
+			helper.Cmd("odo", "storage", "delete", helper.RandString(5), "--context", commonVar.Context, "-f").ShouldFail()
+		})
 
-			volumesMatched := 0
+		When("ephemeral is set to true in preference.yaml and doing odo push", func() {
+			BeforeEach(func() {
+				helper.Cmd("odo", "preference", "set", "ephemeral", "true").ShouldPass()
+				helper.Cmd("odo", "push", "--context", commonVar.Context).ShouldPass()
+			})
 
-			// check the volume name and mount paths for the containers
-			deploymentName, err := util.NamespaceKubernetesObject(cmpName, "app")
-			Expect(err).To(BeNil())
-			volNamesAndPaths := commonVar.CliRunner.GetVolumeMountNamesandPathsFromContainer(deploymentName, "runtime", commonVar.Project)
-			volNamesAndPathsArr := strings.Fields(volNamesAndPaths)
-			for _, volNamesAndPath := range volNamesAndPathsArr {
-				volNamesAndPathArr := strings.Split(volNamesAndPath, ":")
+			It("should not create a pvc to store source code", func() {
+				// Verify the pvc size
+				PVCs := commonVar.CliRunner.GetAllPVCNames(commonVar.Project)
+
+				Expect(len(PVCs)).To(Equal(0))
+				output := commonVar.CliRunner.GetVolumeNamesFromDeployment(cmpName, "app", commonVar.Project)
+				value, ok := output["odo-projects"]
+				Expect(ok).To(BeTrue())
+				Expect(value).Should(Equal(("emptyDir")))
+			})
+		})
 
-				for i, storageName := range storageNames {
-					if strings.Contains(volNamesAndPathArr[0], storageName) && volNamesAndPathArr[1] == pathNames[i] {
-						volumesMatched++
+		When("ephemeral is set to false in preference.yaml", func() {
```

Comment:
To be consistent with the previous one:

```suggestion
		When("ephemeral is set to false in preference.yaml and doing odo push", func() {
```

Reply:
Done